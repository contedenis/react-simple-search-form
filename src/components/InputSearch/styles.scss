.input {
  border: 0;
  height: 50px;
  padding: 0 0 0 10px;
  width: 100%;
}