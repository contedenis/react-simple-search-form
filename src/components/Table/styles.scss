.data-table {
  background-color: white;
  text-align: center;

  &__placeholder {
    animation: card-appear 1s ease;
    color: black;
    height: 520px;
    text-align: center;

    @keyframes card-appear {
      0%  {
        opacity: 0;
        transform: scale(0.8);
      }
      100%  {
        opacity: 1;
        transform: scale(1);
      }
    }
  }
}