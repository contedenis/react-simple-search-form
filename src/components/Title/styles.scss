.title {
  color: #F2F1EF;
  font-size: 20px;
  font-weight: bold;
  margin: 0 0 20px 0;
}