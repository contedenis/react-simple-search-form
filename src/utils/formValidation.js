function isValid(value, validate) {
  const regexp = new RegExp(validate);
  const isValid = regexp.test(value);

  return isValid;
}

export {
  isValid,
};
