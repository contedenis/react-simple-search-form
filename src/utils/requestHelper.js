import axios from 'axios';

const FOOBALL_PLAYERS_API = {
  url: 'https://football-players-b31f2.firebaseio.com/',
};

const footballPlayersApi = () => {
  const instance = axios.create({
    baseURL: FOOBALL_PLAYERS_API.url,
  });
  return instance;
};

const footballPlayersApiInstance = footballPlayersApi();

export {
  footballPlayersApiInstance,
};
