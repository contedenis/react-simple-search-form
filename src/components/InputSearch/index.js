// @ packages
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { noop } from 'lodash';
import { Icon, Input } from 'semantic-ui-react';

// @ own
import './styles.scss';

function InputSearch({
  className,
  max,
  min,
  name,
  onChange,
  onDelete,
  pattern,
  placeholder,
  type,
  value,
}) {
  return (
    <Input
      className={cn('input', className)}
      icon={<Icon name="delete" link onClick={() => onDelete(name)} />}
      maxLength={max}
      minLength={min}
      onChange={e => onChange(e.target.value)}
      pattern={pattern}
      placeholder={placeholder}
      type={type}
      value={value}
    />
  );
}

InputSearch.defaultProps = {
  className: '',
  max: null,
  min: null,
  name: '',
  onChange: noop,
  onDelete: noop,
  pattern: '',
  placeholder: 'Type here..',
  type: 'search',
  value: '',
};

InputSearch.propTypes = {
  className: PropTypes.string,
  max: PropTypes.number,
  min: PropTypes.number,
  name: PropTypes.string,
  onChange: PropTypes.func,
  onDelete: PropTypes.func,
  pattern: PropTypes.string,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
};

export default InputSearch;
