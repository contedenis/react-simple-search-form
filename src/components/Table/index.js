// @ packages
import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'semantic-ui-react';

// @ own
import './styles.scss';

function DataTable({ tableHeaders, tableCell }) {
  return (
    <Table unstackable striped className="data-table">
      <Table.Header>
        <Table.Row>
          {tableHeaders && tableHeaders.map((item, key) => (
            <Table.HeaderCell key={key}>{item}</Table.HeaderCell>
          ))}
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {tableCell.length > 0 ? (
          tableCell.map((item, key) => (
            <Table.Row key={key}>
              <Table.Cell>{item.name}</Table.Cell>
              <Table.Cell>{item.position}</Table.Cell>
              <Table.Cell>{item.nationality}</Table.Cell>
              <Table.Cell>{item.dateOfBirth}</Table.Cell>
            </Table.Row>
          ))
        ) : (
          <Table.Row className="data-table__placeholder">
            <Table.Cell colSpan="4">
              Oops, it seems that we have not found players.
            </Table.Cell>
          </Table.Row>
        )}
      </Table.Body>
    </Table>
  );
}

DataTable.propTypes = {
  tableCell: PropTypes.array.isRequired,
  tableHeaders: PropTypes.array.isRequired,
};

export default DataTable;
