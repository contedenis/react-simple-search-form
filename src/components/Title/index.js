import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

import './styles.scss';

function Title({ children, className }) {
  return (
    <h3 className={cn('title', className)}>{children}</h3>
  );
}

Title.defaultProps = {
  className: '',
};

Title.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Title;
