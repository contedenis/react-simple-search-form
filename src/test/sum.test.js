// Link.react.test.js
/* eslint-disable */
import React from 'react';
import Enzime, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import Title from '../components/Title';
import Table from '../components/Table';
import InputSearch from '../components/InputSearch';
import { isValid } from '../utils/formValidation';

Enzime.configure({ adapter: new Adapter() });

describe('<InputSearch />', () => {
  test('change value correctly', () => {
    let name = 'J';

    function onChangeName(e) {
      if (isValid(e, '^[a-z A-Z]*$')) {
        name = e;
      }
    }

    function onDelete(value) {
      if (value === 'name') {
        name = '';
      }
    }

    const wrapper = mount(
      <InputSearch
        onChange={onChangeName}
        onDelete={onDelete}
        type="text"
      />,
    );

    wrapper.setProps({ name: 'name', value: name });

    const value = wrapper.find('input').props().value;
    expect(value).toEqual('J');
    expect(wrapper).toMatchSnapshot();
  });

  test('calls the onChange function', () => {
    let value = '';
    function onChangeName(e) {
      if (isValid(e, '^[a-z A-Z]*$')) {
        value = e;
      }
    }
    const wrapper = mount(
      <InputSearch
        type="text"
        onChange={onChangeName}
      />,
    );

    wrapper.find('input').simulate('change', {
      target: { value: 'D' }
    });
    wrapper.update();
    wrapper.setProps({ value: value });
    const value1 = wrapper.find('input').props().value;
    expect(value1).not.toBe('J');
    expect(wrapper).toMatchSnapshot();
  });

  test('calls the onDelete function', () => {
    let value = 'name';
    function onDelete(valueProps) {
      if (valueProps === 'name') {
        value = '';
      }
    }
    const wrapper = mount(
      <InputSearch
        type="text"
        onDelete={onDelete}
        name="name"
        value="name"
      />,
    );

    const inputValue = wrapper.find('input').props().value;
    expect(inputValue).toBe('name');
    expect(wrapper).toMatchSnapshot();

    const iconProps = wrapper.find('i');
    iconProps.simulate('click');
    wrapper.update();
    wrapper.setProps({ value: value });

    const inputValueUpdated = wrapper.find('input').props().value;
    expect(inputValueUpdated).not.toBe('name');
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Title />', () => {
  test('render children', () => {
    const wrapper = mount(
      <Title>Hello world!</Title>,
    );
    
    const children = wrapper.find('h3').props().children;
    expect(children).toBe('Hello world!');
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Table />', () => {
  test('rendering of header values', () => {
    const TABLE_HEADERS = [
      'Player',
      'Position',
      'Nationality',
      'Age',
    ];

    const wrapper = mount(
      <Table tableHeaders={TABLE_HEADERS} tableCell={[]} />,
    );

    const listOfth = wrapper.find('th').getElements();
    expect(listOfth[0].props.children).toBe("Player");
    expect(listOfth[1].props.children).toBe("Position");
    expect(listOfth[2].props.children).toBe("Nationality");
    expect(listOfth[3].props.children).toBe("Age");
    expect(wrapper).toMatchSnapshot();
  });

  test('rendering of cell values', () => {
    const TABLE_CELLS = [
      {
        name: 'Romelu Lukaku',
        position: 'Centre-Forward',
        nationality: 'Belgium',
        dateOfBirth: '1993-05-13'
      },
    ];

    const wrapper = mount(
      <Table tableHeaders={[]} tableCell={TABLE_CELLS} />,
    );

    const listOfTr = wrapper.find('tr').getElements();
    const listOfTd = listOfTr[1].props.children.map(node => node.props.children);
    expect(listOfTd).toStrictEqual(['Romelu Lukaku', 'Centre-Forward', 'Belgium', '1993-05-13']);
    expect(wrapper).toMatchSnapshot();
  });

  test('rendering of empty values', () => {
    const wrapper = mount(
      <Table tableHeaders={[]} tableCell={[]} />,
    );

    const listOfTd = wrapper.find('td').getElements();
    expect(listOfTd[0].props.children).toBe("Oops, it seems that we have not found players.");
    expect(wrapper).toMatchSnapshot();

    const listOfTr = wrapper.find('tr').getElements();
    expect(listOfTr[1].props.className).toBe('data-table__placeholder');
    expect(wrapper).toMatchSnapshot();
  });
});