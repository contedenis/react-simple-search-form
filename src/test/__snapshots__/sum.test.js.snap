// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch /> calls the onChange function 1`] = `ReactWrapper {}`;

exports[`<InputSearch /> calls the onDelete function 1`] = `ReactWrapper {}`;

exports[`<InputSearch /> calls the onDelete function 2`] = `ReactWrapper {}`;

exports[`<InputSearch /> change value correctly 1`] = `ReactWrapper {}`;

exports[`<Table /> rendering of cell values 1`] = `ReactWrapper {}`;

exports[`<Table /> rendering of empty values 1`] = `ReactWrapper {}`;

exports[`<Table /> rendering of empty values 2`] = `ReactWrapper {}`;

exports[`<Table /> rendering of header values 1`] = `ReactWrapper {}`;

exports[`<Title /> render children 1`] = `ReactWrapper {}`;
